### PR TITLE
[IN PROGRESS] Upsell: Add form options 

### PR DESCRIPTION
**Pull request by norrise120**
Currently in progress! I would like feedback on my approach for structuring the form prop and adding it to the code. Once I get some feedback here, I'll make design updates and add tests!

### CODE REVIEW

File: packages/gestalt/src/Upsell.js [hunk 1]
```diff
@@ -19,8 +20,23 @@ import {
   type DismissButtonType,
 } from './commonTypes.js';
 
+type TextFieldProps = {|
```
Comment (norrise120) [inline packages/gestalt/src/Upsell.js:23]: Looking for feedback on this and the added UpsellForm!

**Comment by AlbertCarreras** [comment]
HIgh-level discussion here: https://www.loom.com/share/381989997428459e9b260c8aa2805394 @norrise120

**Comment by dangerismycat** [comment]
Totally agree with @AlbertCarreras that primary/secondary don't make sense in this context — he's spot-on about the lack of hierarchy. I'm not sold on the need for something like `Upsell.FormField` though…I _think_ I'd be fine with a `formFields` array prop and an `onFormSubmit` function prop. Sub-components are typically used where more flexibility is needed in the implementation, but we actually don't need much flexibility here: we'll show some form fields (I assume there should be a limit on how many), and we'll show a Submit button next to them. (Would there ever need to be a Cancel button as well?)

What do you think, @norrise120 ?

**Comment by norrise120** [comment]
Thanks for the recommendations! I think you both make very great points. I'll play around with the code a bit and see what works best!

> we actually don't need much flexibility here: we'll show some form fields (I assume there should be a limit on how many), and we'll show a Submit button next to them. (Would there ever need to be a Cancel button as well?)

That's absolutely correct! I think we want to be super limited here in what we allow. Design allows for one or two forms and then one button. Designs currently do not include a cancel button and I don't think at this time we would want to add that functionality.

**Comment by dangerismycat** [comment]
> Design allows for one or two forms and then one button. Designs currently do not include a cancel button and I don't think at this time we would want to add that functionality.

Ok, makes sense. I get the inclination to use an object instead of an array then, though I still agree that the primary/secondary hierarchy doesn't fit. I think we're probably fine using an array that we'll only use the first two items of, provided we communicate that in the docs. I'm not sure if it's possible to lock down the max number of items in an array via Flow, but if nothing else we can add a lint rule in Pinboard to alert devs that they're not going to see all their form fields if they had more than two.

**Comment by AlbertCarreras** [comment]
> Totally agree with @AlbertCarreras that primary/secondary don't make sense in this context — he's spot-on about the lack of hierarchy. I'm not sold on the need for something like `Upsell.FormField` though…I _think_ I'd be fine with a `formFields` array prop and an `onFormSubmit` function prop. Sub-components are typically used where more flexibility is needed in the implementation, but we actually don't need much flexibility here: we'll show some form fields (I assume there should be a limit on how many), and we'll show a Submit button next to them. (Would there ever need to be a Cancel button as well?)
> 
> What do you think, @norrise120 ?

I agree it might be overkilling it. I still think we should internally discuss when we prefer one vs the other

image
vs 
image

**Comment by norrise120** [comment]
@AlbertCarreras @dangerismycat Updated per our discussions! Let me know if this structure looks better or if we should go the Upsell.FormField route! If we're good with this approach, I can work on matching to design and fixing responsiveness.

**Comment by norrise120** [comment]
Because we decided to use a different approach, I'm closing this PR in favor of https://github.com/pinterest/gestalt/pull/1396

**Pull request closed without merging**